.github/workflows/rpk-build.yml

```diff
@@ -8,30 +8,124 @@
 # by the Apache License, Version 2.0
 
 name: Go
-on: [push, pull_request]
+on:
+  push:
+    tags: 'release-*'
+    branches:
+      - '*'
+  pull_request:
+    branches:
+      - main
+      - dev
 
 jobs:
   build:
-    name: Build
+    name: Test and Build
     runs-on: ubuntu-latest
     steps:
 
+    - name: Check out code
+      uses: actions/checkout@v2
+
     - name: Set up Go
       uses: actions/setup-go@v2
       with:
-        go-version: ^1.15
+        go-version: 1.13.6
       id: go
 
-    - name: Check out code into the Go module directory
-      uses: actions/checkout@v2
+    - name: Set up Node
+      uses: actions/setup-node@v2-beta
+      with:
+        node-version: '12.16.1'
+
+    - name: Install JS
+      working-directory: src/js
+      run: |
+        npm install
+
+    - name: Generate WASM deps
+      working-directory: src/js
+      run: |
+        npm run generate:serialization
+        npm run build:public
+        sed -i 's/%CONTENT%/"$(cat vectorizedDependency.js)"/g' ../go/rpk/pkg/cli/cmd/wasm/template/vectorized_dependency.go
 
     - name: Test
+      working-directory: src/go/rpk/
       run: go test -v ./...
+
+    - name: Build Linux
+      working-directory: src/go/rpk/
+      run: |
+        set -e
+        pkg='vectorized/pkg/cli/cmd/version'
+        tag=$(echo ${{ github.ref }} | sed 's;refs/tags/;;g')
+
+        mkdir linux
+        GOOS=linux GOARCH=amd64 go build -ldflags "-X ${pkg}.version=${tag} -X ${pkg}.rev=${{ github.sha }}" -o ./linux ./...
+
+    - name: Build MacOS
       working-directory: src/go/rpk/
+      run: |
+        set -e
+        pkg='vectorized/pkg/cli/cmd/version'
+        tag=$(echo ${{ github.ref }} | sed 's;refs/tags/;;g')
+
+        mkdir darwin
+        GOOS=darwin GOARCH=amd64 go build -ldflags "-X ${pkg}.version=${tag} -X ${pkg}.rev=${{ github.sha }}" -o ./darwin ./...
 
-    - name: Build
-      run: go build -o . -v ./...
+    - name: Package
       working-directory: src/go/rpk/
+      run: |
+        zip -j rpk-linux-amd64.zip ./linux/rpk
+        zip -j rpk-darwin-amd64.zip ./darwin/rpk
 
-    - name: Publish
-      run: echo "TODO"
+    - name: Upload build artifacts
```

Comment:
Optional but this step can also be guarded with : `if: startsWith(github.ref, 'refs/tags/release-')`